dev/prepare-release.sh

```diff
@@ -39,6 +39,6 @@ new_development_version="$release_version-SNAPSHOT"
 tag="apache-parquet-$release_version-rc$2"
 
 ./mvnw release:clean
-./mvnw release:prepare -DskipTests -Darguments=-DskipTests -Dtag="$tag" "-DreleaseVersion=$release_version" -DdevelopmentVersion="$new_development_version"
+./mvnw release:prepare -Pvector-plugins -DskipTests -Darguments=-DskipTests -Dtag="$tag" "-DreleaseVersion=$release_version" -DdevelopmentVersion="$new_development_version"
```

Comment:
does this command perform compile and upload jars? `vector-plugins` requires JDK 17 to build

Reply:
Ah, that's a good catch!

Reply:
there is an alternative command that applies the version change to all modules (include optional modules)
```
./mvnw versions:set -DgenerateBackupPoms=false -DnewVersion=$release_version
```

Reply:
Or, maybe you can consider releasing Parquet using a higher JDK version, with `-release 8` to ensure Java 8 compatibility.